src/settings_internal.js

```diff
@@ -138,7 +138,7 @@ var TARGET_NOT_SUPPORTED = 0x7FFFFFFF;
 // Wasm backend symbols that are considered system symbols and don't
 // have the normal C symbol name mangled applied (== prefix with an underscore)
 // (Also implicily on this list is any function that starts with string "dynCall_")
-var WASM_SYSTEM_EXPORTS = ['setTempRet0', 'getTempRet0', 'stackAlloc', 'stackSave', 'stackRestore'];
+var WASM_SYSTEM_EXPORTS = ['stackAlloc', 'stackSave', 'stackRestore'];
```

Comment:
how does this relate to the PR?

Reply:
This is an interesting one that had be puzzled for quite a while.   Its pretty hairy.

These two functions (setTempRet0 and getTempRet0) are odd because they exist as JS runtime functions (without a leading `_` but they also exist as JS library functions with the same name that wrap the runtime functions (but with a leading `_`).   

They are never implemented in native code, always in JS, so that also makes them outliers in this list.   The other 3 are exported by native code and that idea here is that unlike all other symbols we do not mangle them but make them available to JS without the leading underscore.

In the dynamic linking code when we do symbol lookup we have a final fallback of:

```
    if (!sym) {                                                                  
      sym = Module[asmjsMangle(symName)];                                        
    } 
```

Without the the change above this results in a lookup of `Module["setTempRet0"]` rather than `Module["_setTempRet0"]` for these symbols.   In -O0 builds this ends up hitting:

```
if (!Object.getOwnPropertyDescriptor(Module, "getTempRet0")) Module["getTempRet0"] = function() { abort("'getTempRet0' was not exported. add it to EXPORTED_RUNTIME_METHODS (see the FAQ)") };   
```

And also `_getTempRet0` (the library version) would not be exported because process_dynamic_libs would not mangle it.

The upshot of this change is that native references to setTempRet0 and getTempRet0 will not refer to the library version of these symbols rather than the runtime versions.   I imagine we could perhaps remove the runtime versions completely at some point.

Reply:
I see... that does seem unnecessarily complex. I worry we're missing something, but this should be well-tested so this seems like a good simplification.

Is it possible to make this change in a separate PR, for better bisection? If not doesn't matter.

Reply:
Yup I was thinking the same thing.

Reply:
I found a way to land this change without that part so I can followup later.